Mathlib/FieldTheory/PurelyInseparable/Exponent.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2025 Michal Staromiejski. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michal Staromiejski
+-/
+import Mathlib.FieldTheory.PurelyInseparable.Basic
+
+/-!
+
+# The exponent of purely inseparable extensions
+
+This file defines the exponent of a purely inseparable extension (if one exists) and
+some related results.
+
+Most results are stated using `ringExpChar K` rather than using `[ExpChar K p]` parameter because
+it gives cleaner API. To use the results in a context with `[ExpChar K p]`, consider using
+`ringExpChar.eq K p` for substitution.
+
+## Main definitions
+
+- `IsPurelyInseparable.HasExponent`: typeclass to assert a purely inseparable field extension
+  `L / K` has an exponent, that is a smallest natural number `e` such that
+  `a ^ ringExpChar K ^ e ∈ K` for all `a ∈ L`.
+- `IsPurelyInseparable.exponent`: the exponent of a purely inseparable field extension.
+- `IsPurelyInseparable.elemExponent`: the exponent of an element of a purely inseparable
+  field extension, that is the smallest natural number `e` such that `a ^ ringExpChar K ^ e ∈ K`.
+
+## Tags
+
+purely inseparable
+
+-/
+
+namespace IsPurelyInseparable
+
+variable (K L : Type*)
+
+section Ring
+
+variable [CommRing K] [Ring L] [Algebra K L]
+
+/-- A predicate class on a ring extension saying that there is a natural number `e`
+such that `a ^ ringExpChar K ^ e ∈ K` for all `a ∈ L`. -/
+class HasExponent : Prop where
+  has_exponent : ∃ e, ∀ a, a ^ ringExpChar K ^ e ∈ (algebraMap K L).range
```

Comment:
Please create a constructor for this class that *does* assume `[ExpChar K p]`.

Reply:
> Please create a constructor for this class that _does_ assume `[ExpChar K p]`.

... which I think is to add a `@[mk_iff]` before the `class HasExponent : Prop where`, and add a `hasExponent_iff'` which uses `[ExpChar K p]` (in this case I think `ringExpChar.eq K p ▸` is not convenient).

Reply:
I would prefer if the version that uses `ExpChar` is the main one, without the `'`.